File: cpp/stl/sort.cpp
#include <vector>
#include <algorithm>
#include <functional>


class A
{
 public:
        int mema;
        int memb;
        A(int a,int b){
                mema = a;
                memb = b;
        }
};


int  main()
{
        std::vector<int> vi;
        vi.push_back(1);
        vi.push_back(8);
        vi.push_back(4);
        vi.push_back(1);
        sort( vi.begin(), vi.end(), std::greater<int>());
        
        std::vector<A> va ;
        va.push_back( A(1,2));
        va.push_back( A(7,8));
        va.push_back( A(5,6));
        va.push_back( A(3,4));
		sort( va.begin(),va.end(), greater<A>() );
}

File: ruby/hash/sort.rb
h = { "a" => 20, "b" => 30, "c" => 10  }
h.sort {|a,b| a[1]<=>b[1]}.each do |k,v|
  print k,",",v,"\n"
end
   #=> [["c", 10], ["a", 20], ["b", 30]]
